src/lib.rs

```diff
@@ -210,10 +210,14 @@ pub enum Format {
     Repeat1(Box<Format>),
     /// Repeat a format an exact number of times
     RepeatCount(Expr, Box<Format>),
-    /// Restrict a format to a sub-stream of a given number of bytes
-    Slice(Expr, Box<Format>),
-    /// Matches a format at a byte offset relative to the current stream position
-    WithRelativeOffset(Expr, Box<Format>),
+    /// Matches a format without consuming the input
+    WithInput(Box<Format>),
+    /// Takes a given number of bytes from the start of the input, matching
+    /// the format against those bytes
+    TakeBytes(Expr, Box<Format>),
+    /// Drops a given number of bytes from the start of the input, matching
+    /// the format against the remaining input
+    DropBytes(Expr, Box<Format>),
```

Comment:
`DropBytes` seems unnecessarily parallel to `TakeBytes`, seems like it doesn't need the Format argument and could just skip N bytes? Whereas `TakeBytes` does require the Format argument.

Reply:
Oh right, `TakeBytes` can be used to implement something like `DropBytes`?
```text
def drop-bytes (n : Size) (f : Format) :=
  map _.data {
    skipped <- take-bytes n {},
    data <- f,
  };
```
Hadn’t thought of that! 🤔